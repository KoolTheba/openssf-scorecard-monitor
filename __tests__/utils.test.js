const { validateDatabaseIntegrity, validateScopeIntegrity } = require('../src/utils')
const { database, scope } = require('../__fixtures__')

describe('Utils', () => {
  describe('validateDatabaseIntegrity', () => {
    it('Should manage a full database', () => {
      expect(() => validateDatabaseIntegrity(database.fullDatabase)).not.toThrow()
    })
    it('Should manage an empty database', () => {
      expect(() => validateDatabaseIntegrity(database.emptyDatabase)).not.toThrow()
    })
  })

  describe('validateScopeIntegrity', () => {
    it('Should manage a big Scope', () => {
      expect(() => validateScopeIntegrity(scope.fullScope)).not.toThrow()
    })
    it('Should manage a empty Scope', () => {
      expect(() => validateScopeIntegrity(scope.emptyScope)).not.toThrow()
    })
  })
})
