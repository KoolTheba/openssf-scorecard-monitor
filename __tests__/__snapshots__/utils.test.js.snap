// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils generateReportContent Should render template with scores and title 1`] = `
"# OpenSSF Scorecard Report

## Summary

| Repository | Commit | Score | Date | Difference | Report Link | StepSecurity Link |
| -- | -- | -- | -- | -- | -- | -- |
| [fake-org/fake-repo](https://github.com/fake-org/fake-repo) | [846b3dd](https://github.com/fake-org/fake-repo/commit/846b3ddb5f75d95235e94d9eb52e920f4a067338) | 10 | 2023-02-20 | 5 | [Full Report](https://deps.dev/project/github/fake-org%2Ffake-repo) | [Fix it](http://app.stepsecurity.io/securerepo?repo=fake-org/fake-repo) |

_Report generated by [UlisesGascon/openssf-scorecard-monitor](https://github.com/UlisesGascon/openssf-scorecard-monitor)._"
`;

exports[`Utils generateReportContent Should render template with scores only 1`] = `
"
| Repository | Commit | Score | Date | Difference | Report Link | StepSecurity Link |
| -- | -- | -- | -- | -- | -- | -- |
| [fake-org/fake-repo](https://github.com/fake-org/fake-repo) | [846b3dd](https://github.com/fake-org/fake-repo/commit/846b3ddb5f75d95235e94d9eb52e920f4a067338) | 10 | 2023-02-20 | 5 | [Full Report](https://deps.dev/project/github/fake-org%2Ffake-repo) | [Fix it](http://app.stepsecurity.io/securerepo?repo=fake-org/fake-repo) |

_Report generated by [UlisesGascon/openssf-scorecard-monitor](https://github.com/UlisesGascon/openssf-scorecard-monitor)._"
`;

exports[`Utils generateReportContent Should render template with title only 1`] = `
"# OpenSSF Scorecard Report

## Summary


_Report generated by [UlisesGascon/openssf-scorecard-monitor](https://github.com/UlisesGascon/openssf-scorecard-monitor)._"
`;
